spring-cloud-stream-binder-jms-activemq/src/main/java/org/springframework/cloud/stream/binder/jms/activemq/ActiveMQQueueProvisioner.java

```diff
@@ -44,18 +149,21 @@ public Destinations provisionTopicAndConsumerGroup(String topicName, String... c
 					 * therefore we must remove '.' from the consumer group name if present.
 					 * For example, anonymous consumer groups are named 'anonymous.*' by default.
 					 */
-					groups[i] = createQueue(topicName, session,	consumerGroupName[i].replaceAll("\\.", "_"));
+					groups[i] = createQueue(topicName, session, consumerGroupName[i].replaceAll("\\.", "_"));
 				}
 			}
 
 			JmsUtils.commitIfNecessary(session);
 			JmsUtils.closeSession(session);
 			JmsUtils.closeConnection(activeMQConnection);
-		} catch (JMSException e) {
-			e.printStackTrace();
+			if (groups != null) {
+				return groups[0];
+			}
 		}
-
-		return new Destinations(topic, groups);
+		catch (JMSException e) {
+			throw new IllegalStateException(e);
```

Comment:
I though we agreed to use a generic `ProvisioningException` here - did I miss a decision change?